spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -921,8 +921,8 @@ class CometSparkSessionExtensions
     override def apply(plan: SparkPlan): SparkPlan = {
 
       // Comet required off-heap memory to be enabled
-      if ("true" != conf.getConfString("spark.memory.offHeap.enabled", "false")) {
-        logInfo("Comet extension disabled because spark.memory.offHeap.enabled=false")
+      if (!isOffHeapEnabled(conf) && !isTesting) {
+        logWarning("Comet extension disabled because spark.memory.offHeap.enabled=false")
```

Comment:
Should we add this to extended explain info?

Reply:
added